doc/mgmt/SONiC_Design_Doc_Unified_FRR_Mgmt_Interface.md

```diff
@@ -0,0 +1,1069 @@
+    * [4.3 REST Get Sequence](#43-rest-get-sequence)
+      * [4.3.1 REST Get Sequence - config only](#431-rest-get-sequence-config-only)
+      * [4.3.2 REST Get Sequence - State/Statistics](#432-rest-get-sequence-state-statistics)
+  * [5 Error Handling](#6-error-handling)
+  * [6 Serviceability and Debug](#7-serviceability-and-debug)
+  * [7 Warm Boot Support](#8-warm-boot-support)
+  * [8 Scalability](#9-scalability)
+  * [9 Unit Test](#10-unit-test)
+  * [APPENDIX](#APPENDIX)
+
+
+## List of Tables
+[Table 1: Abbreviations](#table-1-abbreviations)
+
+## Revision
+| Rev |     Date    |       Author       | Change Description                |
+|:---:|:-----------:|:------------------:|-----------------------------------|
+| 0.1 | 09/25/2019  | Karthikeyan Arumugam | Initial version                 |
+| 0.2 | 10/30/2019  | Venkatesan Mahalingam | Config DB schema changes |
+| 0.3 | 11/20/2019  | Venkatesan Mahalingam | Added various fields in config DB |
+| 0.4 | 12/18/2019  | Venkatesan Mahalingam | Addressed the comments on error handling and method of testing |
+
+## About this Manual
+This document provides general information about the implementation of Extended Unified Configuration and Management framework support for FRR-BGP feature in SONiC.
+## Scope
+This document describes the high level design of FRR-BGP Extended Unified Configuration and Management feature.
+
+## Definition/Abbreviation
+
+### Table 1: Abbreviations
+| **Term**                 | **Meaning**                         |
+|--------------------------|-------------------------------------|
+| FRR                      | Free Range Routing Stack            |
+| CVL                      | Config Validation Library           |
+| VRF                      | Virtual routing forwarding          |
+| RIB                      | Routing Information Base            |
+| PBR                      | Policy based routing                |
+| NBI                      | North Bound Interface               |
+
+
+
+# 1 Feature Overview
+
+This feature extends and provides unified configuration and management capability for FRR-BGP features used in SONiC. This allows the user to configure & manage FRR-BGP using SONiC Management Framework with Open Config data models via REST, gNMI and also provides access via SONiC Management Framework CLI as well.
+
+
+## 1.1 Requirements
+
+
+### 1.1.1 Functional Requirements
+
+  1. Extend Unified mode for full FRR-BGP config and management in SONiC
+  2. Extend sonic-cfggen, bgpcfgd and integrate with FRR-BGP for features supported in SONiC
+  3. Support for retrieval of FRR-BGP state and statistics information
+  4. For backward compatibility retain access to FRR UI (vtysh) for managing features that are NOT in conflict with SONiC features
+
+### 1.1.2 Configuration and Management Requirements
+
+1. Support Open Config data models for BGP config and Management
```

Comment:
Can you please put urls to the models you want to support?

Reply:
There are bunch of URLs (100+) that we support for BGP configs, adding every URL will affect the readability of this document, rather, referring the YANG itself will help to know what containers/leafs...we support from OC for BGP configs.

Reply:
Sure